server/lib/device/device.saveLastStateChanged.js

```diff
@@ -0,0 +1,44 @@
+const db = require('../../models');
+const logger = require('../../utils/logger');
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../utils/constants');
+
+/**
+ * @description Save new device feature state in DB.
+ * @param {Object} deviceFeature - A DeviceFeature object.
+ * @example
+ * saveLastValueChanged({
+ *   id: 'fc235c88-b10d-4706-8b59-fef92a7119b2',
+ *   selector: 'my-light'
+ * });
+ */
+async function saveLastStateChanged(deviceFeature) {
```

Comment:
I'm still very sceptical with this function, I don't think this is useful. 

The `newStateEvent` function is doing the exact same job. 

I guess the only thing you want is to avoid saving multiple time device state in case it's the same value, and I don't think this is something linked to this PR.

This is another topic!